chainstate/src/detail/mod.rs

```diff
@@ -281,101 +287,89 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
     /// a BlockError and return it.
     /// On each iteration, before doing anything else, call `on_new_attempt`
     /// (this can be used for logging).
-    fn with_rw_tx<MainAction, OnNewAttempt, OnDbErr, Res>(
+    // FIXME: move it somewhere else (i.e. into a different file; it could a part
```

Comment:
If you can find a better abstraction, feel free. But I'm not for having multiple files that work on the same struct/enum/primitive. It's all about good abstractions, not just blindly separating files (this is how the Query interface was created, btw, there was a need to move things that are often used + require r/o support).